skada/_utils.py

```diff
@@ -59,11 +59,18 @@ def check_X_y_domain(
     X = check_array(X, input_name='X', allow_nd=allow_nd)
     y = check_array(y, force_all_finite=True, ensure_2d=False, input_name='y')
     check_consistent_length(X, y)
-    if sample_domain is None and allow_auto_sample_domain:
+    if sample_domain is None and not allow_auto_sample_domain:
```

Comment:
We need to make sure that all of those clauses are properly covered with unit tests. I guess just updating `check_X_y_domain_exceptions` with a bunch of correct and incorrect inputs would do.